subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/version/CompilerMetaDataProvider.java

```diff
@@ -20,5 +20,12 @@
 import java.util.List;
 
 public interface CompilerMetaDataProvider {
+
     GccVersionResult getGccMetaData(File gccBinary, List<String> additionalArgs);
+
+    /**
+     * Returns true if the implementation is Clang, false if GCC.
+     */
+    boolean isClang();
```

Comment:
This looks horrible. Can we at least use an enum here?

Reply:
Or maybe the method needs to be renamed to something more descriptive?

Reply:
Better now?